robot/web/stream/server.go

```diff
@@ -372,6 +374,58 @@ func (server *Server) GetStreamOptions(
 	}, nil
 }
 
+// SetStreamOptions implements part of the StreamServiceServer. It sets the resolution of the stream
+// to the given width and height.
+func (server *Server) SetStreamOptions(
+	ctx context.Context,
+	req *streampb.SetStreamOptionsRequest,
+) (*streampb.SetStreamOptionsResponse, error) {
+	if req.Name == "" {
+		return nil, errors.New("stream name is required in request")
+	}
+	if req.Resolution == nil {
+		return nil, fmt.Errorf("resolution is required to resize stream %q", req.Name)
+	}
+	if req.Resolution.Width <= 0 || req.Resolution.Height <= 0 {
+		return nil, fmt.Errorf(
+			"invalid resolution to resize stream %q: width (%d) and height (%d) must be greater than 0",
+			req.Name, req.Resolution.Width, req.Resolution.Height,
+		)
+	}
+	server.mu.Lock()
+	defer server.mu.Unlock()
+	err := server.resizeVideoSource(req.Name, int(req.Resolution.Width), int(req.Resolution.Height))
+	if err != nil {
+		return nil, fmt.Errorf("failed to resize video source for stream %q: %w", req.Name, err)
+	}
+	streamState, ok := server.nameToStreamState[req.Name]
+	if !ok {
+		return nil, fmt.Errorf("stream state not found with name %q", req.Name)
+	}
+	err = streamState.Resize()
+	if err != nil {
+		return nil, fmt.Errorf("failed to resize stream %q: %w", req.Name, err)
+	}
```

Comment:
Could we move this logic into `resizeVideoSource`?

Reply:
Yep -- moved all resize logic into `resizeVideoSource`.